engine/src/main/java/com/datatorrent/stram/api/StreamingContainerUmbilicalProtocol.java

```diff
@@ -341,6 +341,29 @@ public String toString()
     }
   }
 
+  enum ShutdownType
+  {
+    /**
+     * Container is not scheduled for shutdown.
+     */
+    NO_SHUTDOWN,
+    /**
+     * SOFT_SHUTDOWN, when this value is set all input operators in the
+     * container terminates and send END_STREAM tuple to downstream
+     * operators. This type of shutdown make sure that data emitted by upstream
+     * operator before shutdown are processed by downstreams operator before
+     * they terminate.
+     */
+    SOFT_SHUTDOWN,
+    /**
+     * HARD_SHUTDOWN, when this value is set then all operators in the containers
+     * are terminated. If downstream container receives this request before upstream
+     * operator then downstream operators may not process all the data emitted by
+     * upstream operators.
+     */
+    HARD_SHUTDOWN,
```

Comment:
This is for maintaining current behaviour of shutdown, as there is a differece that after SOFT_SHUTDOWN application can not be restarted from last state as the operators are removed from PhysicalPlan. HARD_SHUTDOWN application can be relaunched again.